### PR TITLE
Fix AddProvider to add providers to existing loggers

### PR DESCRIPTION
**Pull request by BrennanConroy**

### CODE REVIEW

File: test/Microsoft.Extensions.Logging.Test/LoggerTest.cs [hunk 1]
```diff
@@ -101,6 +101,22 @@ public void Log_AggregatesExceptionsFromMultipleLoggers()
...
+        public void LoggerCanGetProviderAfterItIsCreated()
+        {
+            // Arrange
+            var store = new List<string>();
+            var loggerFactory = new LoggerFactory();
+            var logger = loggerFactory.CreateLogger("Test");
+            loggerFactory.AddProvider(new CustomLoggerProvider("provider1", ThrowExceptionAt.None, store));
+
+            // Act
+            logger.LogInformation("Hello");
```
Comment (kichalla) [inline test/Microsoft.Extensions.Logging.Test/LoggerTest.cs]: Is this fix related to logs not being visible from Kestrel? Have you tested this fix with an end-to-end application?

Comment (BrennanConroy) [inline test/Microsoft.Extensions.Logging.Test/LoggerTest.cs]: Yes, tested with the Mvc Sample and Kestrel sample
